pkg/registry/service/rest.go

```diff
@@ -277,8 +277,8 @@ var _ = rest.Redirector(&REST{})
 
 // ResourceLocation returns a URL to which one can send traffic for the specified service.
 func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
-	// Allow ID as "svcname" or "svcname:port".
-	svcName, portStr, valid := util.SplitPort(id)
+	// Allow ID as "svcname", "svcname:port", or "svcname:port:scheme".
```

Comment:
You can have //-- it would just be encoded as %2f%2f (how sad that I know the code without looking it up). Which is ugly, hence my second suggestion.

If there's no documentation, I guess that's a problem bigger than this PR...